kafka/metrics.go

```diff
@@ -83,44 +105,88 @@ func newKgoHooks(mp metric.MeterProvider, namespace, topicPrefix string) (*metri
 		// Consumer
 		messageFetched: messageFetchedCounter,
 		messageDelay:   messageDelayHistogram,
+
+		topicAttributeFunc: topicAttributeFunc,
 	}, nil
 }
 
 func formatMetricError(name string, err error) error {
 	return fmt.Errorf("cannot create %s metric: %w", name, err)
 }
 
-// OnProduceRecordUnbuffered records the number of produced / failed messages.
+// HookProduceBatchWritten is called when a batch has been produced.
+func (h *metricHooks) OnProduceBatchWritten(_ kgo.BrokerMetadata,
+	topic string, partition int32, m kgo.ProduceBatchMetrics,
+) {
+	attrs := make([]attribute.KeyValue, 0, 6) // Preallocate 5 elements.
```

Comment:
```suggestion
	attrs := make([]attribute.KeyValue, 0, 6)
```

comment doesn't match reality, and I think the code speaks for itself